ckan/lib/authenticator.py

```diff
@@ -37,3 +32,13 @@ def authenticate(
             return user.name
         signals.failed_login.send(login)
         return None
+
+
+class CkanAuthenticator(object):
+
+    @classmethod
+    def __call__(cls):
+        for item in plugins.PluginImplementations(plugins.IAuthenticator):
+            return item
+
+        return UsernamePasswordAuthenticator()
```

Comment:
Can we add some documentation of this class and why it is needed?

Also, this implementation will only accept one plugin in our stack implementing `plugins.IAuthenticator`. Is that the expected behavior?

Reply:
I will document it for sure, first, I needed to be sure if I m doing the right thing here.


> Also, this implementation will only accept one plugin in our stack implementing `plugins.IAuthenticator`. Is that the expected behavior?

The expected behavior as @smotornyuk suggested, is that you can set up a custom authenticator from your extensions.